pkg/server/structlogging/hot_ranges_log_test.go

```diff
@@ -101,72 +89,50 @@ func TestHotRangesStats(t *testing.T) {
-	tableDesc := desctestutils.TestingGetPublicTableDescriptor(
-		tc.Server(0).DB(), keys.SystemSQLCodec, "test", "foo")
-	tc.SplitTable(t, tableDesc, []serverutils.SplitPoint{
-		{TargetNodeIdx: 1, Vals: []interface{}{100}},
-		{TargetNodeIdx: 1, Vals: []interface{}{120}},
-		{TargetNodeIdx: 1, Vals: []interface{}{140}},
-		{TargetNodeIdx: 1, Vals: []interface{}{160}},
-		{TargetNodeIdx: 1, Vals: []interface{}{180}},
-		{TargetNodeIdx: 2, Vals: []interface{}{200}},
-		{TargetNodeIdx: 2, Vals: []interface{}{220}},
-		{TargetNodeIdx: 2, Vals: []interface{}{240}},
-		{TargetNodeIdx: 2, Vals: []interface{}{260}},
-		{TargetNodeIdx: 2, Vals: []interface{}{280}},
-	})
 
-	// query table
-	for i := 0; i < 300; i++ {
-		db := tc.ServerConn(0)
-		sqlutils.MakeSQLRunner(db).Query(t, `SELECT * FROM test.foo`)
+	leakChecker := leaktest.AfterTest(t)
+	logInterceptor := log.InterceptWith(ctx, spy)
+	stopper := tc.Stopper()
+	teardown := func() {
+		stopper.Stop(ctx)
+		sc.Close(t)
+		logInterceptor()
+		leakChecker()
 	}
 
-	// Skip node 1 since it will contain many more replicas.
-	// We only need to check nodes 2 and 3 to see that the nodes are logging their local hot ranges.
-	rangeIDs := make(map[int64]struct{})
-	for _, i := range []int{1, 2} {
-		spy.Reset()
-		ts := tc.ApplicationLayer(i)
-		structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
-		structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, time.Second)
-		structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
-
-		testutils.SucceedsSoon(t, func() error {
-			logs := spy.Logs()
-			if len(logs) < 5 {
-				return errors.New("waiting for hot ranges to be logged")
-			}
+	ts := tc.ApplicationLayer(0)
+	return ts, spy, teardown
+}
 
-			return nil
-		})
-		structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, 1*time.Hour)
+// TestHotRangesStatsTenants tests that hot ranges stats are logged per node.
+// The test will ensure each node contains 5 distinct range replicas for hot
+// ranges logging. Each node should thus log 5 distinct range ids.
+func TestHotRangesStats(t *testing.T) {
+	skip.UnderRace(t)
+	ctx := context.Background()
+	ts, spy, teardown := setupHotRangesLogTest(t, ctx)
+	defer teardown()
+
+	structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
+	structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, time.Millisecond)
+	structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
+
+	testutils.SucceedsSoon(t, func() error {
+		logs := spy.Logs()
 
-		// Get first 5 logs since the logging loop may have fired multiple times.
-		// We should have gotten 5 distinct range ids, one for each split point above.
-		logs := spy.Logs()[:5]
+		// Depend on a range which we don't exist to go anywhere.
```

Comment:
I don't quite understand this comment.

Reply:
Would it make sense to say "Look for a range, which we always expect to exist in the system."

Reply:
Yeah that sgtm! Ah, did you mean `expect` in the original comment?

Reply:
Ahh yes! yeah - the original wording doesn't make too much sense.